File: selection_sort.cpp
#include<iostream>
using namespace std;
void selectionSort(int arr[],int n)
{
     for (int i=0; i<=n-2; i++)
     {
        int mini=i;
        for(int j=i;j<=n-1;j++)
        {
            if(arr[j]<arr[mini])
            {
                mini=j;
            }
        }
        int temp=arr[mini];
        arr[mini]=arr[i];
        arr[i]=temp;
     }
}

int main()
{
    int n;
    int arr[]={13,46,24,52,20,9};
    selectionSort(arr,n);
    for (int i=0;i<n; i++)
    {
        cout<<arr[i]<<"\t";
    }
    return 0;
}

File: selection_sort.py
def selection_sort(arr):
    n = len(arr)
    for i in range(n - 1):
        mini = i
        for j in range(i + 1, n):
            if arr[j] < arr[mini]:
                mini = j
        temp=arr[mini]
        arr[mini]=arr[i]
        arr[i]=temp


    print("After selection sort:")
    for i in range(n):
        print(arr[i], end=" ")
    print()

arr = [13, 46, 24, 52, 20, 9]
n = len(arr)
selection_sort(arr)
